src/components/molecules/Visualizer/hooks.ts

```diff
@@ -52,6 +62,25 @@ export default ({
 }) => {
   const engineRef = useRef<EngineRef>(null);
 
+  const [overriddenSceneProperty, overrideSceneProperty] = useState({});
+
+  const handleScenePropertyOverride = useCallback((property: any) => {
+    overrideSceneProperty(p => mergeProperty(p, property));
+  }, []);
+
+  const mergedSceneProperty = useMemo(() => {
+    if (!overriddenSceneProperty || typeof overriddenSceneProperty !== "object")
+      return sceneProperty;
+    return mergeWith(sceneProperty, overriddenSceneProperty, (obj: any, other: any) => {
+      if (Array.isArray(obj) && Array.isArray(other)) {
+        const merged = merge({}, keyBy(obj, "id"), keyBy(other, "id"));
+        return values(merged);
+      } else {
+        return merge({}, obj, other);
+      }
+    });
+  }, [sceneProperty, overriddenSceneProperty]);
```

Comment:
```suggestion
  const mergedSceneProperty = useMemo(() => {
    return mergeProperty(sceneProperty, overriddenSceneProperty)
  }, [sceneProperty, overriddenSceneProperty]);
```

Reply:
I believe this works if I make the above suggested change, but doesn't taking in to consideration what I mention above. I'll wait for the above reply before making any changes here.